proto/ibc/core/channel/v1/channel.proto

```diff
@@ -122,6 +137,10 @@ message Packet {
   ibc.core.client.v1.Height timeout_height = 7 [(gogoproto.nullable) = false];
   // block timestamp (in nanoseconds) after which the packet times out
   uint64 timeout_timestamp = 8;
+  // which IBC version to use to commit this packet
+  IBCVersion protocol_version = 9;
```

Comment:
The idea here being rather than introspecting the channel-id, we have the sender specify a version?

For other reviewers, you should think of the packet type as our go api and the commitment structure as the spec requirement. 2 ibc implementations could use different internal packet types and still be spec compliant